test/tests/functional/pbs_mom_dynamic_resource.py

```diff
@@ -338,8 +349,124 @@ def test_alter_res_value(self):
         self.server.expect(JOB, {'job_state': 'Q', 'comment': c},
                            id=jid, attrop=PTL_AND)
 
+    def test_mom_dyn_res_permissions(self):
+        """
+        Test whether mom rejects the mom dynamic resource script when the
+        permission of the script are open to write for others and group
+        """
+        scr_body = ['echo "10"', 'exit 0']
+        home_dir = os.path.expanduser("~")
+        fp = self.du.create_temp_file(body=scr_body, dirname=home_dir)
+        # Add to filenames for cleanup
+        self.filenames.append(fp)
+
+        mom_config_str = '!' + fp
+        self.scheduler.set_sched_config({'mom_resources': 'foo'},
+                                        validate=True)
+        self.scheduler.add_resource('foo')
+        self.mom.add_config({'foo': mom_config_str})
+
+        # give write permission to group and others
+        self.du.chmod(path=fp, mode=0766)
+        self.check_access_log(fp)
+
+        # give write permission to group
+        self.du.chmod(path=fp, mode=0764)
+        self.check_access_log(fp)
+
+        # give write permission to others
+        self.du.chmod(path=fp, mode=0746)
+        self.check_access_log(fp)
+
+        # give write permission to user only
+        self.du.chmod(path=fp, mode=0744)
+        self.check_access_log(fp, exist=False)
+
+        # Create a script in home directory which has more open privileges
```

Comment:
You are creating the script in a directory that's inside the home directory. I think the comment should just mention that you are creating the script in a directory that is not secure